src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs

```diff
@@ -278,5 +278,141 @@ public void ResolveSdkTiming ()
 			Assert.AreEqual (validateJavaVersion.MinimumRequiredJdkVersion, "1.8", "MinimumRequiredJdkVersion should be 1.8");
 			Directory.Delete (Path.Combine (Root, path), recursive: true);
 		}
+
+		static object [] TargetFrameworkPairingParameters = new [] {
+			//We support 28, but only 27 is installed
+			new object [] {
+				"Older API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"28",   //supportedApiLevel
+			},
+			//28 is installed but we only support 27
+			new object [] {
+				"Newer API Installed", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				null,   //userSelected
+				"27",   //androidApiLevel
+				"27",   //androidApiLevelName
+				"v8.1", //targetFrameworkVersion
+				"27",   //supportedApiLevel
+			},
+			//A paired downgrade to API 26
+			new object [] {
+				"Paired Downgrade", //description
+				// androidSdks
+				new [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "27", Level = 27, Name = "Oreo", FrameworkVersion = "v8.1", Stable = true },
+				},
+				// targetFrameworks
+				new ApiInfo [] {
+					new ApiInfo () { Id = "26", Level = 26, Name = "Oreo", FrameworkVersion = "v8.0", Stable = true },
+					new ApiInfo () { Id = "28", Level = 28, Name = "P",    FrameworkVersion = "v9.0", Stable = true },
+				},
+				null,   //userSelected
+				"26",   //androidApiLevel
+				"26",   //androidApiLevelName
+				"v8.0", //targetFrameworkVersion
+				"28",   //supportedApiLevel
```

Comment:
...again, I don't understand why we want `$(_SupportedApiLevel)` to refer to an API level that we're *not* using. Shouldn't this be 26?